From 2eda2605cfbcf18f22dd6c95e678ed391fccef27 Mon Sep 17 00:00:00 2001
From: Johannes Schindelin <johannes.schindelin@gmx.de>
Date: Fri, 22 Nov 2019 11:20:22 +0100
Subject: [PATCH 25/N] Set up a GitHub Action to keep in sync with Cygwin

This will help us by automating an otherwise tedious task.

Signed-off-by: Johannes Schindelin <johannes.schindelin@gmx.de>
---
 .github/workflows/sync-with-cygwin.yml | 21 +++++++++++++++++++++
 1 file changed, 21 insertions(+)
 create mode 100644 .github/workflows/sync-with-cygwin.yml

diff --git a/.github/workflows/sync-with-cygwin.yml b/.github/workflows/sync-with-cygwin.yml
new file mode 100644
index 0000000..defcdf8
--- /dev/null
+++ b/.github/workflows/sync-with-cygwin.yml
@@ -0,0 +1,21 @@
+name: sync-with-cygwin
+
+# File: .github/workflows/repo-sync.yml
+
+on:
+  schedule:
+  - cron:  "42 * * * *"
+jobs:
+  repo-sync:
+    runs-on: ubuntu-latest
+    steps:
+    - name: Fetch Cygwin's latest master and tags
+      run: |
+        git init --bare
+        # Potentially use git://sourceware.org/git/newlib-cygwin.git directly, but GitHub seems more reliable
+        git fetch https://github.com/cygwin/cygwin master:refs/heads/cygwin/master 'refs/tags/*:refs/tags/*'
+    - name: Push to our fork
+      env:
+        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      run: |
+        git push https://$GITHUB_ACTOR:$GITHUB_TOKEN@github.com/$GITHUB_REPOSITORY refs/heads/cygwin/master 'refs/tags/*:refs/tags/*'
-- 
2.31.0

